Training24/Test.c

```diff
@@ -1,19 +1,172 @@
+         printf ("Invalid choice. Please enter 1, 2, or 3.\n");
+         continue;
+      }
+      switch (choice) {
+      case 1:
+         TestUserPhrase ();
+         break;
+      case 2:
+         TestUserInteger ();
+         break;
+      case 3:
+         printf ("Exiting program.\n");
+         return 0;
+      }
+   }
+   return 0;
+}
+
+void PhraseTableHeader () {
+   printf ("+-------------------------------------+-------------------+\n");
+   printf ("|         Phrase - Test Case          |       Result      |\n");
+   printf ("+-------------------------------------+-------------------+\n");
+}
+
+void PhraseTableRow (const char* input, const char* result) {
+   printf ("| %-35s | %-26s |\n", input, result);
+   printf ("+-------------------------------------+-------------------+\n");
+}
+
+void IntegerTableHeader () {
+   printf ("+-----------------------+--------------------------+-------------------+\n");
+   printf ("|  Integer - Test Case  |        Output            |       Result      |\n");
+   printf ("+-----------------------+--------------------------+-------------------+\n");
+}
+
+void IntegerTableRow (int number, const char* outputStr, bool isPalindrome) {
+   const char* resultStr;
+   if (strcmp (outputStr, "Integer Overflow") == 0) resultStr = YELLOW "Failed to process" RESET;
+   else resultStr = isPalindrome ? GREEN "It's a Palindrome" RESET : RED "Not a Palindrome" RESET;
+   printf ("| %-21d | %-24s | %-26s |\n", number, outputStr, resultStr);
+   printf ("+-----------------------+--------------------------+-------------------+\n");
+}
+
+
+void TestPhrasePalindromes () {
+   PhraseTableHeader ();
+   const char* phrases[] = { "Race!Car", "hello", "D", "Don't nod", "A man, a Plan, a canal, Panama",
+      " ","22 / 02 / 2022", "12345654321" };
+   for (int i = 0; i < sizeof (phrases) / sizeof (phrases[0]); i++) {
+      int isPhrasePalindrome = CheckPhrasePalindrome (phrases[i]);
+      PhraseTableRow (phrases[i], isPhrasePalindrome ? GREEN "It's a Palindrome" RESET : RED "Not a Palindrome" RESET);
+   }
+   printf ("\n\n");
+}
+
+void TestReverseNumbers () {
+   IntegerTableHeader ();
+   int testCases[] = { 212, -121, 123421, -9876789, 2147447412, 2147483647, -2147483648LL, 999999999 };
+   for (int i = 0; i < sizeof (testCases) / sizeof (testCases[0]); i++) {
+      int number = testCases[i];
+      if (number < 0) IntegerTableRow (number, "Invalid(Negative number)", false);
+      else {
+         int reversedNumber;
+         bool overflow = ReverseAndCheckOverflow (number, &reversedNumber);
+         char outputStr[256];
+         if (overflow) IntegerTableRow (number, "Integer Overflow", false);
+         else {
+            sprintf (outputStr, "%d", reversedNumber);
+            bool isPalindrome = (number == reversedNumber);
+            IntegerTableRow (number, outputStr, isPalindrome);
+         }
+      }
+   }
+}
+
+void clearInputBuffer () {
```

Comment:
Function name starts with an uppercase letter. Kindly go through the code once before pushing to avoid these basic mistakes.